code/game/machinery/washing_machine.dm

```diff
@@ -57,161 +103,170 @@
+							break
+				if(istype(O, /obj/item/clothing/accessory/scarf)) // scarves
+					for(var/T in typesof(/obj/item/clothing/accessory/scarf))
+						var/obj/item/clothing/accessory/scarf/J = new T
+						if((wash_color == J.item_color) && J.dyeable)
+							qdel(O)
+							newcontents += J
+							break
+				if(istype(O, /obj/item/bedsheet)) // bedsheets
+					for(var/T in typesof(/obj/item/bedsheet))
+						var/obj/item/bedsheet/J = new T
+						if(wash_color == J.item_color)
+							qdel(O)
+							newcontents += J
+							break
+				if(istype(O, /obj/item/storage/wallet/color)) // only for wallets from the arcade
+					for(var/T in typesof(/obj/item/storage/wallet/color))
+						var/obj/item/storage/wallet/color/J = new T
+						if(wash_color == J.item_color)
+							qdel(O)
+							newcontents += J
+							break
+				if(istype(O, /obj/item/toy/carpplushie)) // carp plushies! was gonna add for fox plushies but their pathing makes it not an option currently
+					for(var/T in typesof(/obj/item/toy/carpplushie))
+						var/obj/item/toy/carpplushie/J = new T
+						if(wash_color == J.item_color)
+							qdel(O)
+							newcontents += J
+							break
+			for(var/I in 1 to (length(newcontents)))
+				contents += newcontents[I]
 		QDEL_NULL(crayon)
 
-
-	if( locate(/mob,contents) )
-		state = 7
-		gibs_ready = 1
+	running = FALSE
+	if(!full)
+		bloody = FALSE
 	else
-		state = 4
+		playsound(src, 'sound/weapons/jug_filled_impact.ogg', 25)
 	update_icon()
-
-/obj/machinery/washing_machine/verb/climb_out()
-	set name = "Climb out"
-	set category = "Object"
-	set src in usr.loc
-
-	sleep(20)
-	if(state in list(1,3,6) )
-		usr.loc = src.loc
-
+	sleep(5)
+	if(stat & NOPOWER)
+		playsound(src, 'sound/machines/terminal_off.ogg', 45)
+	else
+		playsound(src, 'sound/machines/defib_success.ogg', 25)
 
 /obj/machinery/washing_machine/update_icon()
-	icon_state = "wm_[state][panel]"
+	if(bloody)
+		if(running)
+			state = 8
+		else
+			if(open)
+				state = 6
+			else
+				state = 7
+	else
+		if(running)
+			state = 5
+		else
+			if(full)
+				if(open)
+					state = 3
+				else
+					state = 4
+			else
+				if(open)
+					state = 1
+				else
+					state = 2
+
+	icon_state = "wm_[state]"
+	cut_overlays()
+	if(panel)
+		add_overlay("wires")
+	else
+		if(stat & NOPOWER)
+			if(running)
+				add_overlay("power0")
+		else
+			add_overlay("power1")
 
-/obj/machinery/washing_machine/attackby(obj/item/W as obj, mob/user as mob, params)
-	/*if(istype(W,/obj/item/screwdriver))
-		panel = !panel
-		to_chat(user, "<span class='notice'>you [panel ? </span>"open" : "close"] the [src]'s maintenance panel")*/
+/obj/machinery/washing_machine/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/reagent_containers/spray/cleaner) || istype(W, /obj/item/soap))
+		user.visible_message("<span class='notice'>[user] starts to clean [src].</span>", "<span class='notice'>You start to clean [src].</span>")
+		if(do_after(user, 10 * W.toolspeed, target = src))
```

Comment:
Please don't use magic numbers, something like this instead:
W.toolspeed SECONDS